packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -684,6 +862,67 @@ Future<void> _writeAndroidPluginRegistrant(FlutterProject project, List<Plugin>
   );
 }
 
+/// Generates the Dart plugin registrant, which allows to bind a platform
+/// implementation of a Dart only plugin to its interface.
+/// The new entrypoint wraps [currentMainUri], adds a [_registerPlugins] function,
+/// and writes the file to [newMainDart].
+///
+/// [mainFile] is the main entrypoint file. e.g. /<app>/lib/main.dart.
+///
+/// Returns [true] if it's necessary to create a plugin registrant, and
+/// if the new entrypoint was written to disk.
+///
+/// For more details, see https://flutter.dev/go/federated-plugins.
+Future<bool> generateMainDartWithPluginRegistrant(
+  FlutterProject rootProject,
+  String currentMainUri,
+  File newMainDart,
+  File mainFile,
```

Comment:
You can accept a PackageConfig instead of parsing it again, since the kernel compiler already has one

Reply:
Done!